### PR TITLE
Gh actions master

### PR DESCRIPTION
**Pull request by d-chambers**
### What does this PR do?

Adds workflows for running the default test suite on the supported OS/python version and the network test suite on Linux and one python version. Also updated the build badge. 

### Why was it initiated?  Any relevant Issues?

Requested in #2688.  

### PR Checklist
- [ ] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] If the PR is making changes to documentation, docs pages can be built automatically.
      Just remove the space in the following string after the + sign: "+ DOCS"
- [ ] If any network modules should be tested for the PR, add them as a comma separated list
      (e.g. `clients.fdsn,clients.arclink`) after the colon in the following magic string: "+TESTS:"
      (you can also add "ALL" to just simply run all tests across all modules)
- [ ] All tests still pass.
- [ ] Any new features or fixed regressions are be covered via new tests.
- [ ] Any new or changed features have are fully documented.
- [ ] Significant changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by d-chambers** [comment]
Hmmm... failures are related to basemap's use of the matplotlib cook book.

File: .github/workflows/master_default_tests.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# It simply runs all of ObsPy's default tests on all support OS/python versions
+name: master_default_tests
+on:
+  push:
+    branches:
+      - master
+
+# Set globally scoped environmental variables.
+env:
+  OBSPY_GITHUB_BRANCH: gh-actions
```
Comment (megies) [inline .github/workflows/master_default_tests.yml:11]: will this need to be changed later?

**Comment by megies** [comment]
Theose basemap fails we cant do much about, anyway not related to this PR. There's some other fails so that are troublesome, but again, not related.

Go ahead and merge whenever you think this is ready, we'll only see it in effect after merging anyway.

File: .github/workflows/master_default_tests.yml (hunk 1 above)
Comment (d-chambers) [reply .github/workflows/master_default_tests.yml:11]: You mean if I make a proper version branch for obspy_github_api? Yes that would be good to do :smile:

**Comment by d-chambers** [comment]
Ok. One thing that bothers me is the amount of repetition in these actions. Supposedly the GH action folks are working on something to help this but there isn't a good way to do it yet. I will refactor later when there is a clean way to write more precise GH actions.

**Pull request merged**